### PR TITLE
Fix to work with the new Pythran API

### PR DESCRIPTION
**Pull request by aguinet**
Pythran 0.9.0 has a new API regarding the shape of ndarrays. This PR supports this new API.

This **must not** be merged until Pythran 0.9.0 is out!

### CODE REVIEW

**Comment by aguinet** [comment]
cc @serge-sans-paille

**Comment by serge-sans-paille** [comment]
Thanks @aguinet. I'll update this thread once the new release is out.

**Comment by scoder** [comment]
Rather than breaking what's there and having to wait for users to install what's new, I think it would be better to have something that works with both the older and the new versions. @aguinet @serge-sans-paille, do you think that's possible? We know the Pythran version at translation time, right? Couldn't we base the distinction on that? (Or maybe even something at C++ compile time?)

File: Cython/Compiler/Pythran.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def pythran_type(Ty, ptype="ndarray"):
             ctype = dtype.typedef_cname
         else:
             raise ValueError("unsupported type %s!" % dtype)
-        return "pythonic::types::%s<%s,%d>" % (ptype,ctype, ndim)
+        return "pythonic::types::%s<%s,pythonic::types::pshape<%s>>" % (ptype,ctype, ",".join(("long",)*ndim))
```
Comment (scoder) [inline Cython/Compiler/Pythran.py]: @serge-sans-paille @aguinet
What about this change, is this correct for Pythran 0.8.7 ? Does it work with older Pythran versions or is this a breaking change?

Comment (scoder) [reply Cython/Compiler/Pythran.py]: I changed it to use `Py_ssize_t` and it seems to work for me (on 64 bit Linux).

**Comment by scoder** [comment]
I integrated the changes manually, hopefully in a backwards compatible way.

**Comment by serge-sans-paille** [comment]
> Rather than breaking what's there and having to wait for users to install what's new, I think it would be better to have something that works with both the older and the new versions. @aguinet @serge-sans-paille, do you think that's possible? We know the Pythran version at translation time, right? Couldn't we base the distinction on that? (Or maybe even something at C++ compile time?)

It's posisble to generate different code depending on the detected ``pythran.__version__``. It's a new situation for pythran: there is no API contract on the C++ interface...

@scoder Yes, the change is correct for pythran 0.8.7.

**Pull request closed without merging**